src/charm.py

```diff
@@ -118,6 +120,21 @@ def main(self, event):
                                 "periodSeconds": 30,
                             }
                         },
+                        "volumeConfig": [
```

Comment:
The file could be created directly in the container.  Would you prefer creating it directly?

In absence of a case against, I prefer going through a configmap though because:
* upstream does it (the fewer differences, the easier to reconcile on updates)
* in future, this could be driven by relations that update the configmap without restarting the workload (eg: `juju relate profiles katib` adds the `katib=enabled` to the configmap)

Reply:
When I said "create the files inside this container properly", that was kind of a misnomer. What I meant was to use `volumeConfig` like this:

```python
[
    {
        "name": "namespace-labels-data",
        "mountPath": "/etc/profile-controller",
        "files": [
            {
                "path": "namespace-labels.yaml",
                "content": "katib-metricscollector-injection:      'enabled' ... etc ...",
            },
        ],
    },
]
```

That will actually create a `ConfigMap` as well, it just won't be named `namespace-labels-data`. So we'd get the restartless updates, but this won't work if the service code expects `configmap/namespace-labels-data` to exist in addition to having the data provided in the container already by a different `ConfigMap`.

Reply:
Oh I did not realise that creating a `volumeConfig` with content directly just creates a `configMap` in the background.  That makes sense.  So the answer your question: No, I don't think the profiles workload actually looks for the `namespace-labels-data` configMap, just the mounted content, so this would work.

I think I like using a `configMap` here explicitly still for the same reasons though